File: JULY-2025/29-07-2025/ASCII Range Sum.py
class Solution:
    def asciirange(self, s):
        # code here
        position = {}
        for i in range(len(s)):
            if s[i] not in position:
                position[s[i]] = [i,-1]
            else:
                position[s[i]][1] = i
        
        
        dp = [0] * len(s)
        
        for i in range(len(s)):
            if i > 0:
                dp[i] = dp[i-1] + ord(s[i])
            else:
                dp[i] = ord(s[i])
        
        res =[]
        
        for i,j in position.values():
            if j == -1:
                continue
            
            if dp[j-1] - dp[i] > 0:
                res.append(dp[j-1] - dp[i])
        
        return res

File: JULY-2025/29-07-2025/ASCII Range Sum.cpp
class Solution {
  public:
    vector<int> asciirange(string& s) {
        // code here
        vector<int>ans;
        unordered_map<int,int>mpp, res;
        vector<int>prefixSum(s.size(), 0);
        for(int i = 0;i<s.size();i++){
            int ch = s[i] - 'a';
            if(mpp.find(ch)!=mpp.end()){
                if(res.find(ch)!=res.end())
                 res[ch] = fmax(res[ch], prefixSum[i-1] - prefixSum[mpp[ch]]);
                else 
                    res[ch] = prefixSum[i-1] - prefixSum[mpp[ch]];
            }else{
                mpp[ch] = i;
            }
            int temp = s[i];
            if(i)prefixSum[i] = prefixSum[i-1] + s[i];
            else prefixSum[i] = s[i];
            
        }
        for(auto it : res) if(it.second)ans.push_back(it.second);
        return ans;
    }
};